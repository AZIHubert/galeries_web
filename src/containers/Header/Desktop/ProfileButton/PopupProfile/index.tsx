import * as React from 'react';
import { useSelector } from 'react-redux';

import Image from '#components/Image';
import Text from '#components/Text';

import {
  Gear,
  Ticket,
} from '#ressources/svgComponents';

import {
  profilePictureCurrentSelector,
  userSelector,
} from '#store/selectors';

import {
  Background,
  Button,
  Container,
  Fader,
  InnerContainer,
  Link,
  LogoContainer,
  ProfileImage,
} from './styles';

interface PopupProfileI {
  handleClose: () => void;
  handleOpenTicket: () => void;
  testId?: string;
}

const PopupProfile = ({
  handleClose,
  handleOpenTicket,
  testId,
}: PopupProfileI) => {
  const { croped, pending } = useSelector(profilePictureCurrentSelector);
  const user = useSelector(userSelector);

  return (
    <Fader>
      <Container
        testId={testId}
      >
        <InnerContainer>
          <Link
            onClick={handleClose}
            to='/profile'
          >
            <ProfileImage>
              <Image
                original={croped}
                pending={pending}
              />
            </ProfileImage>
            <div>
              <Text
                color='primary'
                styles={{
                  fontSize: 1,
                }}
                stylesLaptopL={{
                  fontSize: 1.1,
                }}
              >
                { user ? user.userName : 'user name' }
              </Text>
              <Text
                color='primary'
                styles={{
                  fontSize: 0.7,
                }}
                stylesLaptopL={{
                  fontSize: 0.8,
                }}
              >
                  see your profile
              </Text>
            </div>
          </Link>
          <Link
            onClick={handleClose}
            to='/edit'
          >
            <LogoContainer>
              <Gear />
            </LogoContainer>
            <Text
              color='primary'
              styles={{
                fontSize: 0.8,
              }}
              stylesLaptopL={{
                fontSize: 0.9,
              }}
            >
              Edit your informations
            </Text>
          </Link>
          <Button
            onClick={() => {
              handleClose();
              handleOpenTicket();
            }}
          >
            <LogoContainer>
              <Ticket />
            </LogoContainer>
            <div>
              <Text
                color='primary'
                styles={{
                  fontSize: 0.5,
                }}
                stylesLaptopL={{
                  fontSize: 0.65,
                }}
              >
                Share your opinion? Find a bug?
              </Text>
              <Text
                color='primary'
                styles={{
                  fontSize: 0.8,
                }}
                stylesLaptopL={{
                  fontSize: 0.9,
                }}
              >
                send a ticket
              </Text>
            </div>
          </Button>
        </InnerContainer>
        <Background />
      </Container>
    </Fader>
  );
};

export default PopupProfile;
