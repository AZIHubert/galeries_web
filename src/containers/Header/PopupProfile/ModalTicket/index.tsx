import * as React from 'react';
import { useFormik } from 'formik';

import { ticketSchema } from '#helpers/schemas';

interface ModalTicketI {
  loading: boolean;
  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
}

const initialValues = {
  header: '',
  body: '',
};

const ModalTicket = ({ loading, setLoading }: ModalTicketI) => {
  const formik = useFormik({
    initialValues,
    onSubmit: () => {
      if (!loading) { setLoading(true); }
    },
    validateOnChange: false,
    validateOnBlur: true,
    validationSchema: ticketSchema,
  });
  return (
    <div>
      <form onSubmit={formik.handleSubmit}>
        <label
          htmlFor="header"
        >
          title *
        </label>
        <input
          data-testid='headerField'
          disabled={loading}
          id='header'
          name='header'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          type='text'
          value={formik.values.header}
        />
        {formik.errors.header
        && formik.touched.header
        && (
          <div
            data-testid='headerError'
          >
            {formik.errors.header}
          </div>
        )}
        <label
          htmlFor="body"
        >
          body *
        </label>
        <input
          data-testid='bodyField'
          disabled={loading}
          id='body'
          name='body'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          type='text'
          value={formik.values.body}
        />
        {formik.errors.body && formik.touched.body && (
          <div
            data-testid='bodyError'
          >
            {formik.errors.body}
          </div>
        )}
        <div>
          * Required field
        </div>
        <button
          disabled={loading}
          data-testid='submitButton'
          type='submit'
        >
          {loading ? 'loading' : 'send'}
        </button>
      </form>
    </div>
  );
};

export default ModalTicket;
