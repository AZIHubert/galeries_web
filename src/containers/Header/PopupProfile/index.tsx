import * as React from 'react';

import Modal from '#components/Modal';

const PopupProfile = () => {
  const [openModal, setOpenModal] = React.useState<boolean>(false);
  const handleClose = () => {
    setOpenModal(false);
  };
  return (
    <>
      <div
        data-testid='popupProfil'
      >
        <button>
          <img
            alt='profile picture'
          />
          <p>
              userName
          </p>
          <p>
              see your profile
          </p>
        </button>
        <button>
          <img
            alt='edit your information pictogram'
          />
          <p>
              Edit your informations
          </p>
        </button>
        <button
          data-testid='buttonTicketModal'
          onClick={() => setOpenModal((nextState) => !nextState)}
        >
          <img
            alt='send a ticket pictogram'
          />
          <p>
              Share your opinion? Find a bug?
          </p>
          <p>
              send a ticket
          </p>
        </button>
      </div>
      <Modal
        open={openModal}
        handleClose={handleClose}
      >
        <div
          data-testid='ticketModal'
        />
      </Modal>
    </>
  );
};

export default PopupProfile;
