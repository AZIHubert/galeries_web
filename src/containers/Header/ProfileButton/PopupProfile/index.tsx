import * as React from 'react';

import Text from '#components/Text';

import { UserContext } from '#contexts/UserContext';

import defaultProfilePicture from '#ressources/svg/defaultProfilePicture.svg';
import {
  Gear,
  Ticket,
} from '#ressources/svgComponents';

import {
  Background,
  Button,
  Container,
  InnerContainer,
  LogoContainer,
  ProfileImage,
} from './styles';

const PopupProfile = () => {
  const { user } = React.useContext(UserContext);
  return (
    <Container
      data-testid='popupProfil'
    >
      <InnerContainer>
        <Button
          borderBottom
        >
          <ProfileImage
            alt='profile picture'
            src={defaultProfilePicture}
          />
          <div>
            <Text
              fontSize={1.2}
            >
              { user ? user.userName : 'user name' }
            </Text>
            <Text
              fontSize={0.8}
            >
              see your profile
            </Text>
          </div>
        </Button>
        <Button
          borderBottom
        >
          <LogoContainer>
            <Gear />
          </LogoContainer>
          <Text>
            Edit your informations
          </Text>
        </Button>
        <Button
          data-testid='buttonTicketModal'
        >
          <LogoContainer>
            <Ticket />
          </LogoContainer>
          <div>
            <Text
              fontSize={0.65}
            >
              Share your opinion? Find a bug?
            </Text>
            <Text>
              send a ticket
            </Text>
          </div>
        </Button>
      </InnerContainer>
      <Background />
    </Container>
  );
};

export default PopupProfile;
