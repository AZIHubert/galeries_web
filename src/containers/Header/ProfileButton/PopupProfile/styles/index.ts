export { default as Background } from './Background';
export { default as Button } from './Button';
export { default as Container } from './Container';
export { default as InnerContainer } from './InnerContainer';
export { default as LogoContainer } from './LogoContainer';
export { default as ProfileImage } from './ProfileImage';
