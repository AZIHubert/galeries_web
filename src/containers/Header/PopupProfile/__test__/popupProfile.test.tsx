import * as React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {
  cleanup,
  fireEvent,
  render,
  screen,
} from '@testing-library/react';

import PopupProfile from '../index';

describe('PopupProfile', () => {
  beforeAll(() => {
    // @ts-ignore
    ReactDOM.createPortal = jest.fn((element) => element);
  });
  afterEach(() => {
    // @ts-ignore
    ReactDOM.createPortal.mockClear();
  });
  afterEach(cleanup);
  it('renders without crashing', () => {
    const tree = renderer.create(<PopupProfile />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should not display ticket modal on mount', () => {
    const { queryByTestId } = render(<PopupProfile />);
    const ticketModal = queryByTestId('ticketModal');
    expect(ticketModal).toBeNull();
  });
  it('open the ticket modal', () => {
    const { getByTestId } = render(<PopupProfile />);
    const buttonTicketModal = getByTestId('buttonTicketModal');
    fireEvent.click(buttonTicketModal);
    const ticketModal = screen.queryByTestId('ticketModal');
    expect(ticketModal).not.toBeNull();
  });
  it('should close the ticket modal', () => {
    const { getByTestId } = render(<PopupProfile />);
    const buttonTicketModal = getByTestId('buttonTicketModal');
    fireEvent.click(buttonTicketModal);
    fireEvent.click(buttonTicketModal);
    const ticketModal = screen.queryByTestId('ticketModal');
    expect(ticketModal).toBeNull();
  });
});
