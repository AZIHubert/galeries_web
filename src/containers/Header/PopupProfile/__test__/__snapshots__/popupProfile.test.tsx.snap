// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupProfile renders without crashing 1`] = `
<div
  data-testid="popupProfil"
>
  <button>
    <img
      alt="profile picture"
    />
    <p>
      userName
    </p>
    <p>
      see your profile
    </p>
  </button>
  <button>
    <img
      alt="edit your information pictogram"
    />
    <p>
      Edit your informations
    </p>
  </button>
  <button
    data-testid="buttonTicketModal"
    onClick={[Function]}
  >
    <img
      alt="send a ticket pictogram"
    />
    <p>
      Share your opinion? Find a bug?
    </p>
    <p>
      send a ticket
    </p>
  </button>
</div>
`;
